### PR TITLE
fix: "Allow Password Change for OAuth Users" setting is not honored in the "Forgot Password" flow

### PR DESCRIPTION
**Pull request by matheusbsilva137**
## Proposed changes (including videos or screenshots)

 - Correctly check for the `Accounts_AllowPasswordChangeForOAuthUsers` in the `users.forgotPassword` endpoint.

## Steps to test or reproduce

1. Go to **Settings > Accounts > Allow Password Change for OAuth Users** and disable this setting.
2. Create a user via LDAP or OAuth;
3. Log out and use the "Forgot your password?" option to reset the password of the user that has just been created;
**Current behavior:** the reset password e-mail is sent (even to LDAP/OAuth users that didn't have a  password before);
**Expected behavior:** the reset password e-mail should not be sent.

## Further comments

[CORE-439](https://rocketchat.atlassian.net/browse/CORE-439)

[CORE-439]: https://rocketchat.atlassian.net/browse/CORE-439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/server/methods/sendForgotPasswordEmail.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ Meteor.methods<ServerMethods>({
 
 		const email = to.trim().toLowerCase();
 
-		const user = await Users.findOneByEmailAddress(email, { projection: { _id: 1 } });
+		const user = await Users.findOneByEmailAddress(email, { projection: { _id: 1, services: 1 } });
```
Comment (KevLehman) [inline apps/meteor/server/methods/sendForgotPasswordEmail.ts:23]: @MarcosSpessatto wdyt of changing findOneByEmail to accept a type parameter, so this projection is properly typed? 😬 

Know its a bit out of the scope, but would be nice to have

Comment (KevLehman) [reply apps/meteor/server/methods/sendForgotPasswordEmail.ts:23]: Or maybe add it as a separate task, that works as well 😬

Comment (MarcosSpessatto) [reply apps/meteor/server/methods/sendForgotPasswordEmail.ts:23]: yup, for a separate task, looks good

Comment (KevLehman) [reply apps/meteor/server/methods/sendForgotPasswordEmail.ts:23]: @matheusbsilva137 , can u create one for that? 😬

**Review by KevLehman — DISMISSED** [review]

**Review by dougfabris — APPROVED** [review]
Approving it on behalf of frontend team!

File: apps/meteor/server/methods/sendForgotPasswordEmail.ts (hunk 1 above)
Comment (matheusbsilva137) [reply apps/meteor/server/methods/sendForgotPasswordEmail.ts:23]: Sure!

**Review by dougfabris — APPROVED** [review]

**Review by KevLehman — APPROVED** [review]

**Pull request merged**